src/test/java/io/netty/incubator/codec/http3/Http3RequestStreamValidationHandlerTest.java

```diff
@@ -461,6 +466,24 @@ private void testHeadersFrame(Http3HeadersFrame headersFrame, Http3ErrorCode cod
         assertEquals(code == null, channel.finishAndReleaseAll());
     }
 
+    private EmbeddedQuicStreamChannel newClientStream(final BooleanSupplier goAwayReceivedSupplier) throws Exception {
+        return newStream(QuicStreamType.BIDIRECTIONAL, new ChannelInitializer<QuicStreamChannel>() {
+            @Override
+            protected void initChannel(QuicStreamChannel ch) {
+                Http3RequestStreamEncodeStateValidator encStateValidator = new Http3RequestStreamEncodeStateValidator();
+                Http3RequestStreamDecodeStateValidator decStateValidator = new Http3RequestStreamDecodeStateValidator();
+                ch.pipeline().addLast(encStateValidator);
+                ch.pipeline().addLast(decStateValidator);
+                ch.pipeline().addLast(newClientValidator(goAwayReceivedSupplier, qpackAttributes, decoder,
+                        encStateValidator, decStateValidator));
```

Comment:
Would there be a case where a user doesn't want to add `encStateValidator` and `decStateValidator` to the pipeline? If so, should we decouple state object from validating handlers? e.g.

```java
Http3RequestStreamEncodeState encState = ...;
Http3RequestStreamDecodeState decState = ...;
ch.pipeline().addLast(encState.toValidator());
ch.pipeline().addLast(decState.toValidator());
ch.pipeline().addLast(....(... encState, decState));
```

Reply:
The problem is that you can not modify state without the handlers as the state does not have mutating methods. So creating a state without the handler is basically the `NO_[DE/EN]CODE_STATE` constants that I have added.